src/audio/types.h

```diff
@@ -34,7 +35,9 @@ QDebug operator<<(QDebug dbg, ChannelLayout arg);
 
 class ChannelCount {
   public:
-    typedef SINT value_t;
+    // Use a native type with more than 8 bits to avoid -Werror=type-limits
+    // errors on comparisons with the min/max constants.
+    typedef uint16_t value_t;
```

Comment:
I don't think that this is a valid reason for using 16 bit here. 
If you try to use a bigger type or just -1 the check will still fail.

The 255 is anyway a artificial number
Pulseaudio for example allows 32 channels.
https://freedesktop.org/software/pulseaudio/doxygen/sample_8h.html

What is the limit of various encoding formats or ffmpeg?

Reply:
16 bit don't hurt. Why change anything besides the type? This could be done later. I'm not going to research which number of channels is desired or appropriate. The comment explains why 16 bits have been chosen. Nothing more to do here.

Reply:
I don't mind keeping the full unit8 range.

I just think that it is wrong to use a bigger type just for the purpose to detect an out of range values that can't happen if you use a uint8.

If you are afraid someone uses a 16 bit value to set this variable you can implement a private or deleted 16 bit setter.

Once the -Wconversion is in place we fail at compile time anyway.